### PR TITLE
cmake: should check for PYTHON3_VERSION_STRING

### PR DESCRIPTION
**Pull request by tchaikov**
it's a regression introduced by c961e002

Signed-off-by: Kefu Chai <kchai@redhat.com>

- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

**Comment by tchaikov** [comment]
nautilus backport is located at #27384

**Comment by tchaikov** [comment]
https://shaman.ceph.com/builds/ceph/wip-cmake-libpython3/dee623aafafeb1061943ac50f1b562039cb8bf31/

**Comment by b-ranto** [comment]
Hmm, discard my previous comment, while the `PYTHON3_VERSION_STRING` is sort of a step in the right direction, we can only use `PYTHON3_VERSION_MAJOR.PYTHON3_VERSION_STRING` when specifying a version for Python3Libs. This PR should fix the issue: https://github.com/ceph/ceph/pull/27385

**Comment by b-ranto** [comment]
OK, so after a (much more) investigation, I believe I have cracked this. The issue is the following:
- we build require python34 (and python34-devel) from epel in the spec file, that is the easiest way to get python3 in rhel7.6 or less
- rhel7.7 introduces python3 package of version 3.6 as a native package (and therefore, it is part of the build root) but we do not require python3-devel so there is no `/usr/lib64/libpython3.6m.so` and therefore `FindPython3Libs` cannot find it and it fails

We must have recently introduced at least a couple of rhel7.7 builders. Therefore, we have two options, now, depending on whether we want to drop the support for rhel 7.6 and less or not.

This PR is the first option -- forcing the version 3.4 of python in rhel 7 for all minor versoins. This patch should work fine for that.

The other option is dropping support for rhel7.6 or less and build requiring python3-devel (and python3-setuptools and python36-Cython) as implemented here:

https://github.com/ceph/ceph-ci/commit/21b6494d3f24437d2b8d8070b0a129df05d4a881

At a certain time, we will likely want to use the native python3 and we will drop the support for pre-rhel7.7 versions. That is when we should apply the patch above. If we don't want to do it, yet then we should apply this PR (well, we will always need the `PYTHON3_VERSION_STRING` patch) until we decide to revert it and apply the patch above.

File: ceph.spec.in [hunk 1]
```diff
@@ -262,15 +262,9 @@ BuildRequires:  redhat-lsb-core
...
-%if 0%{?rhel} == 7
-BuildRequires:	python34-devel
-BuildRequires:	python34-setuptools
-BuildRequires:	python34-Cython
-%else
-BuildRequires:	python3-devel
-BuildRequires:	python3-setuptools
-BuildRequires:	python3-Cython
-%endif
+BuildRequires:	python%{python3_pkgversion}-devel
```
Comment (tchaikov) [inline ceph.spec.in]: macros `%python3_pkgversion` is defined in the minimal buildroot, so it's safe to used it.

**Pull request merged**